specification/elastic/resource-manager/Microsoft.Elastic/preview/2022-05-05-preview/elastic.json

```diff
@@ -0,0 +1,2305 @@
+          "type": "string"
+        }
+      }
+    },
+    "VMIngestionDetailsResponse": {
+      "description": "The vm ingestion details to install an agent.",
+      "type": "object",
+      "properties": {
+        "cloudId": {
+          "description": "The cloudId of given Elastic monitor resource.",
+          "type": "string"
+        },
+        "ingestionKey": {
+          "description": "Ingestion details to install agent on given VM.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "OperationName": {
+      "enum": [
+        "Add",
+        "Delete"
+      ],
+      "type": "string",
+      "description": "Operation to be performed on the given vm resource id.",
+      "title": "OperationName",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "OperationName"
+      }
+    },
+    "VMCollectionUpdate": {
+      "description": "Update VM resource collection.",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "vmResourceId": {
+          "description": "ARM id of the VM resource.",
+          "type": "string"
+        },
+        "operationName": {
+          "description": "Operation to be performed for given VM.",
+          "$ref": "#/definitions/OperationName"
+        }
+      }
+    },
+    "UpgradableVersionsList": {
+      "description": "Stack Versions that this version can upgrade to",
+      "type": "object",
+      "properties": {
+        "currentVersion": {
+          "type": "string",
+          "description": "Current version of the elastic monitor"
+        },
+        "upgradableVersions": {
+          "type": "array",
+          "x-ms-identifiers": [],
+          "description": "Stack Versions that this version can upgrade to",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ElasticTrafficFilterResponse": {
+      "description": "List of elastic traffic filters in the account",
+      "type": "object",
+      "properties": {
+        "rulesets": {
+          "type": "array",
+          "x-ms-identifiers": [],
+          "description": "List of elastic traffic filters in the account",
+          "items": {
+            "$ref": "#/definitions/ElasticTrafficFilter"
+          }
+        }
+      }
+    },
+    "ElasticTrafficFilter": {
+      "description": "Elastic traffic filter object",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of the elastic filter"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the elastic filter"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the elastic filter"
+        },
+        "region": {
+          "type": "string",
+          "description": "Region of the elastic filter"
+        },
+        "type": {
```

Comment:
Since you are not modelling this as an ARM resource , can you please rename these to avoid confusion with ARM resource types ? 

id\name\type etc are top level ARM concepts this is confusing to the end user

Reply:
As mentioned in the previous comments, these changes are already merged in the private swagger repo and in production. The changes were not requested during the previous PR. This will be a breaking change now. Let us know if this is still required.